ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfilesToggleGroup.tsx

```diff
@@ -1,29 +1,115 @@
 
-import { ComplianceProfileScanStats } from 'services/ComplianceResultsStatsService';
+import { ComplianceProfileSummary } from 'services/ComplianceCommon';
+
+const NON_STANDARD_TAB = 'Other';
+
+// Extract unique standards from profiles and add 'Other' standard if there are profiles with no standards
+function getUniqueStandards(profiles: ComplianceProfileSummary[]): string[] {
+    const standards = new Set(
+        profiles.flatMap((profile) => profile.standards.map((standard) => standard.shortName))
+    );
+    if (profiles.some((profile) => profile.standards.length === 0)) {
+        standards.add(NON_STANDARD_TAB);
+    }
+    return Array.from(standards);
+}
+
+function getInitialStandard(
+    profiles: ComplianceProfileSummary[],
+    profileNameParam: string
+): string {
+    const profile = profiles.find((profile) => profile.name === profileNameParam);
+    if (profile && profile.standards.length > 0) {
+        return profile.standards[0].shortName;
+    }
+    return NON_STANDARD_TAB;
+}
+
+function isStandardInProfile(standardShortName: string, profile: ComplianceProfileSummary) {
+    return (
+        profile.standards.some((standard) => standard.shortName === standardShortName) ||
+        (standardShortName === 'Other' && profile.standards.length === 0)
+    );
+}
 
 type ProfilesToggleGroupProps = {
-    profiles: ComplianceProfileScanStats[];
+    profiles: ComplianceProfileSummary[];
     handleToggleChange: (selectedProfile: string) => void;
 };
 
 function ProfilesToggleGroup({ profiles, handleToggleChange }: ProfilesToggleGroupProps) {
     const { profileName: profileNameParam } = useParams();
 
+    const uniqueStandards = useMemo(() => getUniqueStandards(profiles), [profiles]);
+    const initialStandard = useMemo(
+        () => getInitialStandard(profiles, profileNameParam),
+        [profileNameParam, profiles]
+    );
+
+    const [selectedStandard, setSelectedStandard] = useState(initialStandard);
+
+    useEffect(() => {
+        // Sets the selected standard based on the profile name in the URL.
+        // Currently picks the first standard found since no profile should have multiple standards, however
+        // if this changes in the future, we'll want to find all matches and only update selectedStandard if the
+        // current selectedStandard doesn't exist in the match
+        if (profileNameParam) {
+            const standardShortName =
+                profiles.find((profile) => profile.name === profileNameParam)?.standards[0]
+                    ?.shortName || NON_STANDARD_TAB;
+            setSelectedStandard(standardShortName);
+        }
+    }, [profileNameParam, profiles]);
+
+    function handleStandardSelection(standardShortName) {
+        setSelectedStandard(standardShortName);
+        const firstProfileInStandard = profiles.find((profile) =>
+            isStandardInProfile(standardShortName, profile)
+        );
+        if (firstProfileInStandard) {
```

Comment:
Although not necessarily related to rendering issue that Saif noticed, possible follow up for later:
* decide if `find` method is separate source of truth from `filter` method below?
* is there a need to synchronize the change in `selectedStandard`
* is the `if` condition only theoretical? if it is possible, is the component in an inconsistent state?

Reply:
- I want to say that `profiles` are the source of truth. However I think I follow the question, why not just pull the first the first filtered profile from below... 🤔 I'll mess with that bit
- Also a good questions. Without testing myself, it does seem unnecessary to have the `setSelectedStandard` there.
- That's really there to satisfy typescript, there _shouldn't_ be a situation where a standard doesn't have a profile attached to it.

Reply:
> That's really there to satisfy typescript

That is reason enough. Although I thought so, thank you for confirming.